x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersionComponent.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml;
+
+import org.elasticsearch.action.admin.cluster.node.info.ComponentVersionNumber;
+import org.elasticsearch.common.VersionId;
+
+public class MlConfigVersionComponent implements ComponentVersionNumber {
+    @Override
+    public String componentId() {
+        return "MlConfigVersion";
```

Comment:
Should this be CamelCase or snake_case?

Reply:
my 2c: if we are keeping the current key:value for the X content, I would go for snake_case.
Otherwise it does not matter much.